smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/metrics/MetricDecorator.java

```diff
@@ -0,0 +1,39 @@
+package io.smallrye.reactive.messaging.metrics;
+
+import java.util.function.Consumer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.Tag;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+
+import io.smallrye.reactive.messaging.PublisherDecorator;
+
+@ApplicationScoped
+public class MetricDecorator implements PublisherDecorator {
+
+    @Inject
+    private Instance<MetricRegistry> registryInstance;
+
+    @Override
+    public PublisherBuilder<? extends Message> decoratePublisher(PublisherBuilder<? extends Message> publisher,
+            String channelName) {
```

Comment:
the interface use `name` as parameter name, here you use `channelName`, should it be consistent?

Reply:
Yeah, it should, I'll change the interface since I think I don't think it's particularly obvious that it's the name of the channel

Reply:
Interface updated